offlinebans.lua

```diff
@@ -0,0 +1,146 @@
+
+local PLUGIN = PLUGIN
+
+PLUGIN.name = "Offline Character Bans"
+PLUGIN.author = "Gary Tate"
+PLUGIN.description = "Allows staff to ban characters which are not actively loaded on the server."
+PLUGIN.license = [[
+The MIT License (MIT)
+Copyright (c) 2020 Gary Tate
+Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+]]
+
+ix.lang.AddTable("english", {
+	cmdCharBanOfflineDesc = "Ban a character not loaded onto the server.",
+	cmdCharUnbanOfflineDesc = "Unban a character not loaded onto the server.",
+	cmdCharUnbanOfflineNotBanned = "%s is not banned!",
+	cmdCharBanOfflineOverloadedResult = "More than one character matches the parameters.",
+	cmdCharBanOfflineNoResult = "No character matches the parameters.",
+	cmdCharBanOfflineAlreadyBanned = "%s is already banned!",
+	cmdCharBanOfflineSuccess = "%s has offline banned the character %s (%s).",
+	cmdCharUnbanOfflineSuccess = "%s has offline unbanned the character %s (%s).",
+	cmdCharBanOfflineCharacterLoaded = "%s is loaded on the server."
+})
+
+function PLUGIN:IsCharacterLoaded(id)
+	return ix.char.loaded[tonumber(id)]
+end
+
+function PLUGIN:GetCharacter(client, character, steamID64)
+	local data, id
+
+	local query = mysql:Select("ix_characters")
+		query:Select("data")
+		query:Select("id")
+
+		query:Where("schema", Schema.folder)
+		query:Where("name", character)
+		query:Where("steamID", steamID64)
+
+		query:Callback(function(result)
+			if (result) then
+				if (#result == 1) then
+					data, id = result[1].data, result[1].id
+				else
+					client:NotifyLocalized("cmdCharBanOfflineOverloadedResult")
+				end
+			else
+				client:NotifyLocalized("cmdCharBanOfflineNoResult")
+			end
+		end)
+	query:Execute()
+
+	return data, id
```

Comment:
`data` and `id` cannot be returned right away because you need to wait for the query callback to finish in order to get any result from the database, which will be after the function finishes executing.

You'll need to run your own callback function once the query has finished executing rather than returning